tests/unit/hive_metastore/test_locations.py

```diff
@@ -718,3 +719,184 @@ def my_side_effect(path, **_):
+        ),
+    ]
+
+
+def test_mount_dont_list_partitions():
+    client = create_autospec(WorkspaceClient)
+
+    first_folder = FileInfo("dbfs:/mnt/test_mount/entity/", "entity/", 0, "")
+    first_first_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/", "xxx=yyy/", 0, "")
+    first_first_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/1.parquet", "1.parquet", "", "")
+    first_second_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/", "xxx=zzz/", 0, "")
+    first_second_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/1.parquet", "1.parquet", "", "")
+    misplaced_json = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/misplaced.json", "misplaced.json", "", "")
+    first_delta_log = FileInfo("dbfs:/mnt/test_mount/entity/_delta_log/", "_delta_log/", 0, "")
+
+    def my_side_effect(path, **_):
+        if path == "/mnt/test_mount":
+            return [first_folder]
+        if path == "dbfs:/mnt/test_mount/entity/":
+            return [first_delta_log, first_first_partition, first_second_partition]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=yyy/":
+            return [first_first_partition_files, misplaced_json]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=zzz/":
+            return [first_second_partition_files]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = Mounts(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot()
+    assert len(results) == 1
+    assert results[0].table_format == "DELTA"
+    assert results[0].is_partitioned
+
+
+def test_mount_infinite_loop():
+
+    client = create_autospec(WorkspaceClient)
+
+    folder = FileInfo("dbfs:/mnt/test_mount/entity/", "entity/", 0, "")
+    partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/", "xxx=yyy/", 0, "")
+    file = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/document=2023-12-20", "document=2023-12-20", "", "")
+
+    def my_side_effect(path, **_):
+        if path == "/mnt/test_mount":
+            return [folder]
+        if path == "dbfs:/mnt/test_mount/entity/":
+            return [partition]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=yyy/":
+            return [file]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=yyy/document=2023-12-20":
+            return [file]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+
+    original_limit = sys.getrecursionlimit()
+    sys.setrecursionlimit(150)  # Temporarily lower recursion limit to catch errors early
```

Comment:
Make this a fixture

Reply:
this may break other tests, as all of them are executed in multiple threads. not quite happy about this. is there another option?

Reply:
used a pytest fixure.